diff -up ./src/winscard.c.silence ./src/winscard.c
--- ./src/winscard.c.silence	2016-11-07 10:38:45.714708859 -0800
+++ ./src/winscard.c	2016-11-07 10:40:03.152735432 -0800
@@ -359,7 +359,9 @@ LONG SCardConnect(/*@unused@*/ SCARDCONT
 	{
 		if (!(dwStatus & SCARD_PRESENT))
 		{
-			Log1(PCSC_LOG_ERROR, "Card Not Inserted");
+			/* Coolkey does the card connect to see if the card
+			 * is here, don't fill up /var/log because if it */
+			Log1(PCSC_LOG_DEBUG, "Card Not Inserted");
 			return SCARD_E_NO_SMARTCARD;
 		}
 
