diff -up ./src/hotplug_libhal.c.free_property ./src/hotplug_libhal.c
--- ./src/hotplug_libhal.c.free_property	2014-05-09 10:14:24.937048808 -0700
+++ ./src/hotplug_libhal.c	2014-05-09 10:18:46.688048610 -0700
@@ -381,6 +381,9 @@ static void HPAddDevice(LibHalContext *c
 		(void)snprintf(fullname, sizeof(fullname), "%s (%s)",
 			driver->readerName, sSerialNumber);
 		readerTracker[i].fullName = strdup(fullname);
+		if (sSerialNumber) {
+			libhal_free_string(sSerialNumber);
+		}
 	}
 	else
 #endif
@@ -394,7 +397,8 @@ static void HPAddDevice(LibHalContext *c
 		driver->libraryPath, deviceName);
 	if ((SCARD_S_SUCCESS != ret) && (SCARD_E_UNKNOWN_READER != ret))
 	{
-		char *parent, *device_file;
+		char *parent= NULL, *device_file;
+		char *orig_device_file = NULL;
 
 		/* get the parent descriptor, without the '_if0' */
 		parent = libhal_device_get_property_string(ctx, udi,
@@ -407,6 +411,7 @@ static void HPAddDevice(LibHalContext *c
 			"linux.device_file", NULL);
 		if (! device_file)
 			goto error;
+		orig_device_file = device_file;
 
 		/* check the format looks correct */
 #define LIBUSB_HEADER "/dev/bus/usb/"
@@ -439,6 +444,12 @@ error:
 
 			(void)CheckForOpenCT();
 		}
+		if (parent) {
+			libhal_free_string(parent);
+		}
+		if (orig_device_file) {
+			libhal_free_string(orig_device_file);
+		}
 	}
 
 	(void)SYS_MutexUnLock(&usbNotifierMutex);
