diff -up ./src/PCSC/pcsclite.h.in.maxreaders ./src/PCSC/pcsclite.h.in
--- ./src/PCSC/pcsclite.h.in.maxreaders	2012-06-26 11:49:57.000000000 -0700
+++ ./src/PCSC/pcsclite.h.in	2018-05-23 12:52:27.745131988 -0700
@@ -192,7 +192,7 @@ extern const SCARD_IO_REQUEST g_rgSCardT
 
 #define PCSCLITE_VERSION_NUMBER		"@VERSION@"	/**< Current version */
 /** Maximum readers context (a slot is count as a reader) */
-#define PCSCLITE_MAX_READERS_CONTEXTS			16
+#define PCSCLITE_MAX_READERS_CONTEXTS			32
 
 #define MAX_READERNAME			128
 
diff -up ./src/PCSC/pcsclite.h.maxreaders ./src/PCSC/pcsclite.h
--- ./src/PCSC/pcsclite.h.maxreaders	2018-05-23 11:59:28.172147622 -0700
+++ ./src/PCSC/pcsclite.h	2018-05-23 12:52:40.552151891 -0700
@@ -192,7 +192,7 @@ extern const SCARD_IO_REQUEST g_rgSCardT
 
 #define PCSCLITE_VERSION_NUMBER		"1.8.8"	/**< Current version */
 /** Maximum readers context (a slot is count as a reader) */
-#define PCSCLITE_MAX_READERS_CONTEXTS			16
+#define PCSCLITE_MAX_READERS_CONTEXTS			32
 
 #define MAX_READERNAME			128
 
