diff -up ./etc/pcscd.init.in.init-fix ./etc/pcscd.init.in
--- ./etc/pcscd.init.in.init-fix	2012-04-30 09:52:21.164326000 -0700
+++ ./etc/pcscd.init.in	2012-04-30 09:52:51.177236000 -0700
@@ -2,7 +2,7 @@
 #
 # pcscd        Starts the pcscd Daemon
 #
-# chkconfig:   2345 25 88
+# chkconfig:   345 27 88
 # description: The PC/SC smart card daemon is a resource manager for the \
 #              PC/SC lite and Musclecard frameworks.  It coordinates \
 #              communications with smart card readers, smart cards, and \
@@ -15,9 +15,9 @@
 # Provides: pcscd
 # Required-Start: $local_fs $remote_fs $syslog
 # Required-Stop: $local_fs $remote_fs $syslog
-# Should-Start: udev hal openct
-# Should-Stop: udev hal openct
-# Default-Start: 2 3 4 5
+# Should-Start: udev haldaemon openct
+# Should-Stop: udev haldaemon openct
+# Default-Start: 3 4 5
 # Default-Stop: 0 1 6
 # Short-Description: Daemon to access a smart card using PC/SC
 # Description: The PC/SC smart card daemon is a resource manager for the
