diff -up ./etc/pcscd.init.in.clean_sockets ./etc/pcscd.init.in
--- ./etc/pcscd.init.in.clean_sockets	2015-02-25 14:48:13.738001918 -0800
+++ ./etc/pcscd.init.in	2015-02-25 14:57:33.252968575 -0800
@@ -36,6 +36,8 @@ umask 077
 exec=@sbindir_exp@/pcscd
 prog=$(basename $exec)
 lockfile=@localstatedir_exp@/lock/subsys/$prog
+socket=@localstatedir_exp@/run/pcscd.comm
+mapfile=@localstatedir_exp@/run/pcscd.pub
 PCSCD_OPTIONS=
 
 # Source config
@@ -57,6 +59,7 @@ stop() {
     retval=$?
     echo
     [ $retval -eq 0 ] && rm -f $lockfile
+    rm -f $socket $mapfile
     return $retval
 }
 restart() {
